File: backend/app/services/quotes/proposal_template.py
"""Proposal template settings stored in ``workspace.settings``.

Single source of truth for the per-workspace branding + boilerplate rendered on
client proposals. Both the settings API (``PUT/GET .../proposal-template``) and
the public proposal page read through :func:`get_proposal_template`, so an
operator edit in settings is reflected on every proposal with no code change —
the "add to it in the future" contract. Mirrors the read-with-defaults pattern
of :mod:`app.services.sla.speed_to_lead` and the reputation settings.
"""

from app.models.workspace import Workspace
from app.schemas.proposal import ProposalTemplateSettings

# Key under ``workspace.settings`` holding the proposal template config.
SETTINGS_KEY = "proposal_template"


def get_proposal_template(workspace: Workspace) -> ProposalTemplateSettings:
    """Return the proposal template for a workspace (defaults when unset).

    ``business_name`` falls back to the workspace name so a brand-new workspace
    renders a sensible proposal before the operator customizes anything.
    """
    raw = (workspace.settings or {}).get(SETTINGS_KEY, {})
    if not isinstance(raw, dict):
        raw = {}
    template = ProposalTemplateSettings(**raw)
    if not template.business_name:
        template.business_name = workspace.name
    return template

File: frontend/src/lib/api/proposal-template.ts
import { apiGet, apiPut } from "@/lib/api";

// Mirrors backend `app/schemas/proposal.py::ProposalTemplateSettings`. This is
// the per-workspace branding + boilerplate for client proposals; editing it
// re-renders every proposal with no code change (the self-serve layer).
export interface ProposalTemplateSettings {
  business_name: string | null;
  logo_url: string | null;
  brand_color: string;
  accent_color: string;
  business_address: string | null;
  business_phone: string | null;
  business_email: string | null;
  intro: string | null;
  default_terms: string | null;
  footer: string | null;
}

export interface UpdateProposalTemplateRequest {
  business_name?: string | null;
  logo_url?: string | null;
  brand_color?: string;
  accent_color?: string;
  business_address?: string | null;
  business_phone?: string | null;
  business_email?: string | null;
  intro?: string | null;
  default_terms?: string | null;
  footer?: string | null;
}

export const proposalTemplateApi = {
  get: (workspaceId: string): Promise<ProposalTemplateSettings> =>
    apiGet<ProposalTemplateSettings>(
      `/api/v1/settings/workspaces/${workspaceId}/proposal-template`,
    ),

  update: (
    workspaceId: string,
    data: UpdateProposalTemplateRequest,
  ): Promise<ProposalTemplateSettings> =>
    apiPut<ProposalTemplateSettings>(
      `/api/v1/settings/workspaces/${workspaceId}/proposal-template`,
      data,
    ),
};
